src/network/networkd-route.c

```diff
@@ -1254,5 +1291,14 @@ int route_section_verify(Route *route, Network *network) {
                 route->gateway_onlink = true;
         }
 
+        if (route->on_device && route->family != AF_INET &&
+            !in_addr_is_null(route->family, &route->gw) &&
+            !in_addr_is_null(route->family, &route->prefsrc) &&
+            route->src_prefixlen && route->dst_prefixlen)
```

Comment:
Maybe `||` instead of `&&`? Of course except for the first one.

Reply:
yes.  I missed this use case #3068. Thanks.